app/scripts/controllers/actionalGenes.js

```diff
@@ -121,9 +181,30 @@ angular.module('oncokbStaticApp')
             }, []);
         }
 
+        function getAllMainTypes() {
+            var deferred = $q.defer();
+            api.getAllMainTypes()
+                .then(function(response) {
+                    $scope.tumorTypes = $scope.tumorTypes.concat(response.data.filter(function(record) {
+                        return !record.trim().endsWith("NOS");
+                    }).map(function(mainType) {
+                        return {
+                            type: 'Main Type',
+                            name: mainType
+                        };
+                    }));
+                    deferred.resolve();
+                }, function() {
+                    deferred.resolve();
```

Comment:
Did you make that on purpose? Should it be deferred.reject()?

Reply:
Yeah, I don't want to reject the call if the main type endpoint gives error. Just don't need to include the main types in the list.